### PR TITLE
Benchmark : Modify/Add workflows/actions for github-action-benchmark

### PR DESCRIPTION
**Pull request by varun-sundar-rabindranath**
SUMMARY:
Modify/Add Github actions and workflows required for `github-action-benchmark` UI

TEST PLAN:
Manual testing

### CODE REVIEW

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -6,6 +6,14 @@ on:
...
 
     workflow_dispatch:
+      inputs:
+        push_benchmark_results_to_gh_pages:
+          description: "When set to true, the workflow pushes all benchmarking results to gh-pages UI "
+          type: choice
+          options:
+              - 'true'
+              - 'false'
+          default: 'false'
```
Comment (varun-sundar-rabindranath) [inline .github/workflows/nightly.yml:16]: This option is to force-push results to the github branch where the benchmark results are maintained for UI

File: .github/actions/nm-github-action-benchmark/action.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        # Add a commit comment comparing the current benchmark with the previous.
+        comment-always: true
+        # Create an alert when some value has regressed more than 10% 
+        alert-threshold: "10%"
+        # Mark the workflow as a failure when some alert is triggered
+        fail-on-alert: true
+        # Add a commit comment describing what triggered the alert
+        comment-on-alert: true
+        # TODO (varun): Is this a reasonable number ? 
+        max-items-in-chart: 50
```
Comment (varun-sundar-rabindranath) [inline .github/actions/nm-github-action-benchmark/action.yml:52]: This option will kick out values older than 50 entries. i.e. if we only push results on nightly we will only have results for the past 50 days.

File: .github/workflows/nm-benchmark.yml [hunk 3]
```diff
@@ -108,9 +121,81 @@ jobs:
...
+
+      - name: display structure of downloaded files
+        run: ls -R ./downloads
+
+      - name: nm-github-action-benchmark(bigger_is_better.json)
+        uses: ./.github/actions/nm-github-action-benchmark
+        with:
+          gh_action_benchmark_json_file_path:  "downloads/bigger_is_better.json"
+          gh_action_benchmark_tool: "customBiggerIsBetter"
+          # TODO (varun) - Switch this to `nm-gh-pages` after testing
```
Comment (varun-sundar-rabindranath) [inline .github/workflows/nm-benchmark.yml:188]: Make sure to switch this to `nm-gh-pages` before landing.

File: .github/workflows/nm-benchmark.yml [hunk 4]
```diff
@@ -108,9 +121,81 @@ jobs:
...
+          auto_push: ${{ inputs.push_benchmark_results_to_gh_pages }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: nm-github-action-benchmark(smaller_is_better.json)
+        uses: ./.github/actions/nm-github-action-benchmark
+        with:
+          gh_action_benchmark_json_file_path:  "downloads/smaller_is_better.json"
+          gh_action_benchmark_tool: "customSmallerIsBetter"
+          # TODO (varun) - Switch this to `nm-gh-pages` after testing
+          gh_pages_branch: "varun/generate-bench-results-test"
```
Comment (varun-sundar-rabindranath) [inline .github/workflows/nm-benchmark.yml:199]: Make sure to switch this to `nm-gh-pages` before landing.

File: .github/workflows/nm-benchmark.yml [hunk 5]
```diff
@@ -108,9 +121,81 @@ jobs:
...
+          retention-days: 1
+
+  NM_GH_ACTION_BENCHMARK:
+
+    needs: BENCHMARK 
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    permissions:
+      # Permissions required to be able to push to the nm-gh-pages branch 
+      contents: write
```
Comment (varun-sundar-rabindranath) [inline .github/workflows/nm-benchmark.yml:164]: Calling out this permission - This is required for the action to be able to write to `nm-gh-pages` branch. However, it grants write access to the entire repo and not just that branch. I couldn't find a finer grained permission.

**Comment by varun-sundar-rabindranath** [comment]
Land this only after https://github.com/neuralmagic/nm-vllm/pull/122 lands.

File: .github/workflows/nm-benchmark.yml (hunk 5 above)
Comment (varun-sundar-rabindranath) [reply .github/workflows/nm-benchmark.yml:164]: @robertgshaw2-neuralmagic

**Review by robertgshaw2-neuralmagic — APPROVED** [review]

File: .github/workflows/nm-benchmark.yml (hunk 5 above)
Comment (varun-sundar-rabindranath) [reply .github/workflows/nm-benchmark.yml:164]: Resolution : Lets keep it for now. We need to audit this permission at the company level. 
If this is not okay there is another option : 
   There is an option to point the action to a different repo - maybe we create one and have the action do all the write to that repo.

**Pull request merged**